### PR TITLE
Issue #5127 - Milestone 1 realtime events fixes

### LINKED ISSUE #5127: [2D MVP] Milestone 1: Realtime upload fixes
### Description
Product issue:  https://github.com/3drepo/3D-Repo-Product-Team/issues/377.

### Bugs
3 things here:
- [ ] Screen that isn't uploading is getting the new container/processing event, but it never processed the finished event (this may be a race condition on the messages, if it is please ignore)
- [ ] Screen that did the upload is processing all the events, but when it is done it's showing `ok` instead of the name of th e revision
- [ ] The rendering of the processing statuses should be different? (Should be inline with container)

### Related Resources

MockUp : `link to mockup`

### PR DESCRIPTION
**Pull request by sanmont3drepo**
This fixes #5127 

#### Description
- Fixed double entry for revision when you are the one uploading it
- Fixed calibration status being up to date
- Fixed latest revisions being up to date
- Refactored a bit to remove redundant actions in redux

### CODE REVIEW

File: frontend/test/test.helpers.ts [hunk 1]
```diff
@@ -35,41 +40,54 @@ export const spyOnAxiosApiCallWithFile = (api, method) => {
...
+			resolvePromise = null;
 		}
-	), middlewares);
+		
+		return st;
+	}		
+	, middlewares);
 	
-	const waitForActions = (func, waitActions) =>  { 
+	const waitForActions:WaitForActions = (dispatchingfunction, waitActions, debug?: boolean) =>  {
```
Comment (The-Daniel) [inline frontend/test/test.helpers.ts:84]: Is the idea that if we want to debug a test that uses `waitForActions` we append `true` to it?

**Review by The-Daniel — CHANGES_REQUESTED** [review]

File: frontend/test/test.helpers.ts (hunk 1 above)
Comment (sanmont3drepo) [reply frontend/test/test.helpers.ts:84]: Is just to make it do a console log. After doing that I added a way to run it through vscode so maybe this is no longer necessary.

File: frontend/src/v5/helpers/store.helpers.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import { AnyAction, Store } from 'redux';
+
+let store: Store<unknown, AnyAction> = undefined;
+
+export const setStore = (newStore) => store = newStore;
+
+export const dispatch = (action: AnyAction) => store.dispatch(action);
```
[the remark this answers is not in the document]
Comment (sanmont3drepo) [reply frontend/src/v5/helpers/store.helpers.ts:24]: yup, that's the idea. I didnt want to change all the references to this file ATM I would create too much noise on the PR. I just wanted to put this out there so in the near  future we can replace all other references to this one.

File: frontend/src/v5/helpers/store.helpers.ts
[the remark this answers is not in the document]
Comment (sanmont3drepo) [reply frontend/src/v5/helpers/store.helpers.ts:1]: In nature they seem to tackle very different problems. one seems to be more like helpers for the state within the store ( `/v5/store/store.helpers.ts`) and the other more about helpers for using the store. I think renaming one of them would be more appropriate. 
Any suggestions?

[the remark this answers is not in the document]
Comment (sanmont3drepo) [reply frontend/src/v5/helpers/store.helpers.ts:1]: renamed to `redux.helpers.ts`

**Review by The-Daniel — APPROVED** [review]

File: frontend/src/v4/modules/store.ts [hunk 3]
```diff
@@ -22,6 +22,7 @@ import { applyMiddleware, compose, createStore } from 'redux';
 import createSagaMiddleware from 'redux-saga';
```
Comment (Amantini1997) [inline frontend/src/v4/modules/store.ts:22]: I am looking at the same drawings list from 2 different users (A and B). User A changes the latest active revision. For as long as that drawing is not expanded for User B (i.e. for as long and they haven't expanded the revisions list), the revisions updates via WS are not considered 
 
https://github.com/user-attachments/assets/68840cc7-3d4e-4269-a78d-a8a6fadcd040

The same behaviour is observable for containers

File: frontend/src/v5/store/drawings/revisions/drawingRevisions.selectors.ts [hunk 4]
```diff
@@ -18,26 +18,42 @@
...
 
-export const selectRevisionsPending = createSelector(
+export const selectActiveRevisions = createSelector(
 	selectRevisions,
-	selectDrawingById,
-	(revisions, drawing) => revisions.length !== drawing.revisionsCount,
+	(revisions) => revisions.filter((val) => !val.void),
+); 
+
+export const selectLastRevision = createSelector(
```
Comment (Amantini1997) [inline frontend/src/v5/store/drawings/revisions/drawingRevisions.selectors.ts:45]: Can this be renamed to `selectLatestRevision<Name>` or something similar as it doesn't actually return the revision itself, but its name/code?

**Review by Amantini1997 — CHANGES_REQUESTED** [review]

**Review by Amantini1997 — APPROVED** [review]

**Pull request merged**